RIOT-OS/RIOT
# Introduce ATxmega CPU and Boards

### Contribution description

Add ATxmega CPU and 4 boards. This is an attempt to implement the #15703 Feature Request.
The code was based on the @Josar 's [XMEGA](https://github.com/Josar/RIOT/tree/xmega_xtimer) branch. However, it was ported and enhanced on many aspects.

Details:

#### cpu/atxmega

This concentrate all MCU family. This means, all drivers should be put in here. There is no necessity to create an atxmega_common structure.

Drivers:

- CPUID: Full functional, it gets the 11 bytes unique identification from NVM
- NVM: minimal implementation to supply cpuid.
- GPIO: Full functional driver including interrupts.
- PM: Full functional with all 4 MCU levels.
- Timer: Full functional. It support all multiple instances.
- USART: Full functional asynchronous mode. It support all multiple instances.

Interrupt: The XMEGA have a HW multilevel interrupt controller and current version already enable all levels. However, the implementation selects at board configs all interrupts as LOW LEVEL. This means, the behavior will be same as any other MEGA MCU. User can increase the priority but need understand the implications. The Round Robin queue can be enabled at board start-up to avoid starvation from LOW LEVEL interrupts.

SAUL: Supported

#### boards/atxmega

Follow same structure to concentrate all board common code.

### Testing procedure

Tests were conducted on the following boards:
- atxmega-a1-xplained (XMEGA-A1)
- atxmega-a1-xpro (XMEGA-A1U)
- WM-200  (XMEGA-A3U)

Applications:
- examples/saul
- examples/timer_periodic_wakeup
- tests/buttons
- Tutorials/task-01

In general, the attempts were to make sure code not only builds but it run on real hardware.

### Issues/PRs references

Dependencies: #15712

CC: @benpicco @maribu
